### ISSUE TITLE
Error when I use maven plugin version 1.4.2

### ISSUE THREAD

**Issue by apieraf** [dependency-check/DependencyCheck#547]
I have the following error with the version 1.4.2, but I don't know when I can find more information about the error

```
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:1.4.2:check (default-cli) on project myProject: One or more exceptions occured during dependency-check analysis -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :myProject
```

If I change the version to 1.4.0 to execute dependency check, the analysis works fine

Could you help me to use the latest version?

Thanks

**Comment by jeremylong** [comment]
More details about the error can be found by running maven with -X.

Jeremy

**Comment by apieraf** [comment]
The complete error running maven with -X

```
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:1.4.2:check (default-cli) on project myProject: One or more exceptions occured during dependency-check analysis -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.owasp:dependency-check-maven:1.4.2:check (default-cli) on project myProject: One or more exceptions occured during dependency-check analysis
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: One or more exceptions occured during dependency-check analysis
    at org.owasp.dependencycheck.maven.CheckMojo.runCheck(CheckMojo.java:120)
    at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.execute(BaseDependencyCheckMojo.java:423)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    ... 20 more
Caused by: org.owasp.dependencycheck.exception.ExceptionCollection: One or more exceptions occured during dependency-check analysis
    at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:427)
    at org.owasp.dependencycheck.maven.MavenEngine.analyzeDependencies(MavenEngine.java:95)
    at org.owasp.dependencycheck.maven.CheckMojo.runCheck(CheckMojo.java:97)
    ... 23 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :myProject
```

**Comment by jeremylong** [comment]
When running with -X can you find where ODC is running. More info will
likely be there.  I'll have to adjust the final failure message.

Jeremy

**Comment by apieraf** [comment]
Sorry Jeremy, but I can't find more information or files, I only find the file **dc.h2.db**.

**Comment by jeremylong** [comment]
Okay, I've updated some of the error handling/display and this will be included in the next release. In the meantime, if you run your maven build and pipe the output to a text file. You can then open the text file and you should be able to find a line that looks like:

```
[INFO] --- dependency-check-maven:
```

If you search between the above line and the build failure below you should find [ERROR] or [WARN] messages that will provide a better description of what the error is.

```
[INFO] Analysis Complete (3615 ms)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

**Comment by apieraf** [comment]
I found an error, but I use maven to analyze a Java project, no .NET project

```
[WARNING] An error occurred with the .NET AssemblyAnalyzer;
this can be ignored unless you are scanning .NET DLLs. Please see the log for more details.
[DEBUG] Could not execute GrokAssembly Cannot run program "mono": error=2, No such file or directory
[ERROR] Exception occurred initializing Assembly Analyzer.
[DEBUG] 
org.owasp.dependencycheck.exception.InitializationException: An error occurred with the .NET AssemblyAnalyzer
    at org.owasp.dependencycheck.analyzer.AssemblyAnalyzer.initializeFileTypeAnalyzer(AssemblyAnalyzer.java:255)
    at org.owasp.dependencycheck.analyzer.AbstractFileTypeAnalyzer.initialize(AbstractFileTypeAnalyzer.java:160)
    at org.owasp.dependencycheck.Engine.initializeAnalyzer(Engine.java:442)
    at org.owasp.dependencycheck.maven.MavenEngine.initializeAnalyzer(MavenEngine.java:140)
    at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:380)
    at org.owasp.dependencycheck.maven.MavenEngine.analyzeDependencies(MavenEngine.java:95)
    at org.owasp.dependencycheck.maven.CheckMojo.runCheck(CheckMojo.java:97)
    at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.execute(BaseDependencyCheckMojo.java:423)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
    at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.io.IOException: Cannot run program "mono": error=2, No such file or directory
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1047)
    at org.owasp.dependencycheck.analyzer.AssemblyAnalyzer.initializeFileTypeAnalyzer(AssemblyAnalyzer.java:234)
    ... 29 more
Caused by: java.io.IOException: error=2, No such file or directory
    at java.lang.UNIXProcess.forkAndExec(Native Method)
    at java.lang.UNIXProcess.<init>(UNIXProcess.java:186)
    at java.lang.ProcessImpl.start(ProcessImpl.java:130)
    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1028)
    ... 30 more
```

I hope that information can help you to fix the problem

**Comment by apieraf** [comment]
I found the same error when I analyze the project with dependency check v 1.4.0, but in that case, the analysis ends fine

**Comment by jeremylong** [comment]
Thanks for the information - I'll take another look at the .NET analyzers.
For now, you can simply disable the two .NET analyzers by configuring:

```
<nuspecAnalyzerEnabled>false</nuspecAnalyzerEnabled>
<assemblyAnalyzerEnabled>false</assemblyAnalyzerEnabled>
```

If you are not sure where to set this, see example 4 [here](http://jeremylong.github.io/DependencyCheck/dependency-check-maven/index.html).

--Jeremy

**Comment by apieraf** [comment]
Hi Jeremy, 

I can't modify pom.xml file, for this reason, I will continue using v1.4.0

Thanks

**Comment by colezlaw** [comment]
The AssemblyAnalyzer on non-Windows machines will throw an
InitializationException if mono can't be found. In other execution contexts
(CLI, etc.) this doesn't cause ODC to give up entirely, however. Yes -
you'll get the same warning message as above, but as it states, you should
be able to ignore it unless there are .DLL's or .EXE's in the project that
you want analyzed.

As with other analyzers, when one of these happens, it just gets added to a
collection of exceptions that happened during analysis - it's normal to not
be able to open a file, for a file to be formatted goofy, or something like
that. The engine collects all these failures into a single exception which
is thrown and the engine bubbles it up. In the case of the Maven plugin, it
ONLY is supposed to fail the build if that exception collection is marked
as fatal, and Engine doesn't do it - so I'm not sure how isFatal is being
set to 1.

Is it possible that under that stack trace there might be another warning
about not being able to generate the report?

**Comment by jeremylong** [comment]
@colezlaw with the recent changes to exception handling this initialization exception will actually cause the build to fail...  I've got a few ideas to make this work.

**Comment by jeremylong** [comment]
A patch has been written that will suppress the initialization exception on non-windows platforms if 'mono' cannot be found on the path (and path to mono has not been specified). The patch will be included in 1.4.3.

TODOs were added to document the long term enhancement that will eventually be added to ensure this exception is not ignored in the future.
